components/partials/sign_up/UserData.tsx

```diff
@@ -0,0 +1,122 @@
+// Functionality
+import { useAuth } from "hooks/useAuth";
+import { useTranslation } from "next-i18next";
+
+// Form imports
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useForm } from "react-hook-form";
+import * as yup from "yup";
+
+// Components
+import BrInput from "components/brickroom/BrInput";
+
+//
+
+export namespace UserDataNS {
+  export interface FormValues {
+    email: string;
+    name: string;
+    user: string;
+  }
+
+  export interface Props {
+    onSubmit: (data: FormValues) => void;
+  }
+}
+
+//
+
+export default function UserData({ onSubmit }: UserDataNS.Props) {
+  // Loading translations
+  const { t } = useTranslation("signUpProps", { keyPrefix: "UserData" });
+
+  // Getting function that checks for email
+  const { register } = useAuth();
+
+  /* Form setup */
+
+  const defaultValues: UserDataNS.FormValues = {
+    email: "",
+    name: "",
+    user: "",
+  };
+
+  const schema: yup.AnyObjectSchema = yup
+    .object({
+      name: yup.string().required(),
+      user: yup.string().required(),
+      email: yup
+        .string()
+        .email()
+        .required()
+        .test("email-exists", t("email.invalid"), async (value, testContext) => {
+          return await testEmail(value!);
+        }),
+    })
+    .required();
+
+  // This function checks if the provided email exists
+  async function testEmail(email: string) {
+    const result = await register(email, true);
+    return result?.keypairoomServer ? true : false;
```

Comment:
the only syntactic sugar that I accept could be in this case... you can also use 
```js
return !!result?.keypairoomServer
```
even if I love more explicit things like
```js
return Boolean(result?.keypairoomServer)
```

Reply:
Cool, I'll go with the last one. I didn't know about it